talend-component-maven-plugin/src/it/web/.jenkins/Jenkinsfile

```diff
@@ -0,0 +1,322 @@
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+/**
+ * Pod configuration
+ */
+final String tsbiImage = 'jdk11-svc-springboot-builder'
+final String tsbiVersion = '2.9.18-2.4-20220104141654'
+final String stageDefaultContainer = tsbiImage
+final String podLabel = ((String) "tck-api-test-${UUID.randomUUID().toString()}").take(53)
+final String podConfiguration = """
+  apiVersion: v1
+  kind: Pod
+  spec:
+    containers:
+      - name: '${tsbiImage}'
+        image: 'artifactory.datapwn.com/tlnd-docker-dev/talend/common/tsbi/${tsbiImage}:${tsbiVersion}'
+        command: [ cat ]
+        tty: true
+        volumeMounts: [
+          { name: docker, mountPath: /var/run/docker.sock },
+          { name: efs-jenkins-component-runtime-m2, mountPath: /root/.m2/repository },
+          { name: dockercache, mountPath: /root/.dockercache }
+        ]
+        resources: { requests: { memory: 3G, cpu: '2' }, limits: { memory: 8G, cpu: '2' } }
+    volumes:
+      - name: docker
+        hostPath: { path: /var/run/docker.sock }
+      - name: efs-jenkins-component-runtime-m2
+        persistentVolumeClaim:
+              claimName: efs-jenkins-component-runtime-m2
+      - name: dockercache
+        hostPath: { path: /tmp/jenkins/tdi/docker }
+    imagePullSecrets:
+      - name: talend-registry
+"""
+
+/**
+ * Credentials
+ */
+
+final def nexusCredentials = usernamePassword(
+    credentialsId: 'nexus-artifact-zl-credentials',
+    usernameVariable: 'NEXUS_USER',
+    passwordVariable: 'NEXUS_PASSWORD')
+
+final def tenantId_Rd = string(
+    credentialsId: 'component-runtime-api-tenant-id-rd',
+    variable: 'componentRuntimeApiTenantId_Rd'
+)
+
+/**
+ * Options
+ */
+
+final String serverPort = "8081"
+final String serverLog = "component_server.log"
+final String serverTimeout = "1"
+
+// Default value for parameters
+String runtimeVersion = "default"
+String apiTesterEnv = "component_runtime_ci"
+String fileToRun = "tcomp_approved"
+String tenantInstance = "eu"
+/**
+ * The pipeline
+ */
+pipeline {
+  /**
+   * agent
+   */
+  agent {
+    kubernetes {
+      label podLabel
+      yaml podConfiguration
+      defaultContainer stageDefaultContainer
+    }
+  }
+  /**
+   * parameters
+   */
+  parameters {
+    string(
+      name: 'CYCLE',
+      defaultValue: 'default',
+      description: 'Test Cycle Id filled by Test Orchestrator, it is only important for TTO.'
+    )
+    string(
+      name: 'RUNTIME_VERSION',
+      defaultValue: 'default',
+      description: 'Choose the component-runtime.version to be used from Nexus. Keep default to use the pom version (1.45.0 minimum) or an other ex: 1.45.0-SNAPSHOT'
+    )
+    choice(
+      name: 'PROJECT_NAME' ,
+      choices: ['tcomp_approved', 'tcomp_draft'] ,
```

Comment:
I noticed another json file `tcomp_debug_size_issue.json`, is it forgotten,  useless (or more likely debug purpose :) )

Reply:
It is a file created to do research about a strange behaviour found during this task.
I moved it to another branch linked to TAT-1632